user.go

```diff
@@ -371,5 +424,16 @@ func (a *App) SignOut(userID string) (bool, error) {
 		StatusCode: res.StatusCode(),
 		StatusText: res.Status(),
 		ErrorText:  errorText,
+		ErrorCode:  errorCode,
+	}
+}
+
+func networkError(message string) error {
+	return Error{
+		Message:    message,
+		StatusCode: 500,
```

Comment:
not all errors returned from the openapi client will be 500 from our servers. from what i could tell perusing through the codegen'd files, if both `res` and `err` are not `nil` then i think that means it should be 500 (not 100% sure).

if anything, i would rather err on the side of accuracy. so if it couldn't be easily determined i think it would be better not to set these values at all instead of guessing.

Reply:
If the value isn't set, `status code` will get defaulted to zero (which is what currently happens). I added the 500 internal server error to rectify the `status code: 0`. I would argue that a 500 would at least indicate to the dev that the issue is not on their side where as `status code: 0` just looks like a bug.

Reply:
okay 👍 in the major version it would be good to try to throw different kinds of errors then so we aren't filling out "default" values that aren't applicable